pkg/manifests/manifests.go

```diff
@@ -0,0 +1,150 @@
+package registry
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/suse-edge/edge-image-builder/pkg/http"
+	"github.com/suse-edge/edge-image-builder/pkg/image"
+	"go.uber.org/zap"
+	"gopkg.in/yaml.v3"
+)
+
+func GetAllImages(ctx *image.Context) ([]image.ContainerImage, error) {
+	var combinedManifestPaths []string
+	var extractedImagesSet = make(map[string]string)
+
+	if len(ctx.ImageDefinition.Kubernetes.Manifests.URLs) != 0 {
+		downloadDestination := filepath.Join(ctx.BuildDir, "downloaded-manifests")
+		if err := os.MkdirAll(downloadDestination, os.ModePerm); err != nil {
+			return nil, fmt.Errorf("creating %s dir: %w", downloadDestination, err)
+		}
+
+		downloadedManifestPaths, err := downloadManifests(ctx, downloadDestination)
+		if err != nil {
+			return nil, fmt.Errorf("error downloading manifests: %w", err)
+		}
+
+		combinedManifestPaths = append(combinedManifestPaths, downloadedManifestPaths...)
+	}
+
+	localManifestSrcDir := filepath.Join(ctx.ImageConfigDir, "kubernetes", "manifests")
+	localManifestPaths, err := getLocalManifestPaths(localManifestSrcDir)
```

Comment:
Does this even work if the directory doesn't exist or is empty?

Reply:
Updated this based on our conversation.